### PR TITLE
Add unit tests for Vector

### PR DESCRIPTION
**Pull request by ShaneEverittM**
Modify CMake configuration to fetch GTest v14.

### CODE REVIEW

**Review by aoroudjev — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,6 +3,14 @@ project(ds)
 
 set(CMAKE_CXX_STANDARD 17)
 
+include(FetchContent)
+FetchContent_Declare(
+    googletest
+    URL https://github.com/google/googletest/archive/f8d7d77c06936315286eb55f8de22cd23c188571.zip
+)
+set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
```
Comment (aoroudjev) [inline CMakeLists.txt:11]: Are you sure we need this?

File: Tests/VectorTest.cpp [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#include <gtest/gtest.h>
+
+#include <Structures/Vector.h>
+
+TEST(VectorTest, DefaultConstructor) {
+    Vector<int> v;
+    EXPECT_EQ(v.size(), 0);
+}
+
+TEST(VectorTest, PushBack) {
```
Comment (aoroudjev) [inline Tests/VectorTest.cpp:10]: Should we push_back more elements in this test?

File: CMakeLists.txt (hunk 1 above)
Comment (ShaneEverittM) [reply CMakeLists.txt:11]: Yeah, it's necessary for compatiblity with windows.

File: Tests/VectorTest.cpp (hunk 2 above)
Comment (ShaneEverittM) [reply Tests/VectorTest.cpp:10]: No, this exercise both cases for `grow`: 0 => 1 and 1=>2

**Pull request merged**